nym-authenticator-client/src/error.rs

```diff
@@ -42,10 +42,19 @@ pub enum Error {
 
     #[error("unknown authenticator version number")]
     UnsupportedAuthenticatorVersion,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    NoCredentialSpent(InnerError), // This intentionnally doesn't use `from` to avoid random ? operator to land here when they shouldn't
```

Comment:
perhaps put the inner value into named `source` field like in the `CredentialSpent` variant for consistency sake?